src/xdocs/config_coding.xml

```diff
@@ -3623,6 +3623,200 @@ class TestHashCode {
       </subsection>
     </section>
 
+    <section name="MatchXpath">
+      <p>Since Checkstyle 8.39</p>
+      <subsection name="Description" id="MatchXpath_Description">
+        <p>
+          Evaluates Xpath query and report violation on all matching AST nodes. Current check allows
+          user to implement custom checks using Xpath. If Xpath query is not specified explicitly,
+          then check does nothing.
+        </p>
+        <p>
+          Please read more about Xpath syntax at
+          <a href="https://www.w3schools.com/xml/xpath_syntax.asp"> W3Schools Xpath Syntax</a>.
+          Information regarding Xpath functions can be found at
+          <a href="https://developer.mozilla.org/en-US/docs/Web/XPath/Functions">XSLT/XPath
+            Reference</a>. Note, that <b>@text</b> attribute can used only with token types that
+          are listed in
+          <a href="https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java#L101">
+            XpathUtil</a>.
+        </p>
+
+      </subsection>
+
+      <subsection name="Properties" id="MatchXpath_Properties">
+        <div class="wrapper">
+          <table>
+            <tr>
+              <th>name</th>
+              <th>description</th>
+              <th>type</th>
+              <th>default value</th>
+              <th>since</th>
+            </tr>
+            <tr>
+              <td>query</td>
+              <td>Xpath query.</td>
+              <td><a href="property_types.html#String">String</a></td>
+              <td><code>&quot;&quot;</code></td>
+              <td>8.39</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="MatchXpath_Examples">
+        <p>
```

Comment:
Some of this seems to be examples of other checkstyle functions than the check's functions. Not saying it has to be removed, but it seems like this documentation should be somewhere else and we could point to it from here.

Reply:
didn't understand your point, 
do you want to add links to real checks we try to mimic?

Reply:
lets move this to #9074 , it is long awaited page as Xpath is far from easy concept.

Reply:
I was trying to get at was the examples for the check don't show check examples but instead other Checkstyle CLI examples. An example is it is like I give you a manual to operate a boat, and the manual first talks about how to buy a boat when a manual on operating a boat should assume you have said boat.

This is resolved by new issue.